### PR TITLE
Search refactor (3 columns + styling)

### PR DESCRIPTION
**Pull request by Raemon**

### CODE REVIEW

File: packages/lesswrong/components/common/SearchBar.jsx [hunk 1]
```diff
@@ -1,23 +1,56 @@
...
+import { InstantSearch, SearchBox } from 'react-instantsearch/dom';
+import { withStyles } from '@material-ui/core/styles';
+import classNames from 'classnames';
+import Icon from '@material-ui/core/Icon'
+
+const styles = theme => ({
+  searchBar: {
+    display: "inline-block",
+    position: "relative",
+    "& .ais-SearchBox__submit":{
```
Comment (jglamine) [inline packages/lesswrong/components/common/SearchBar.jsx]: Where does this class come from?

Comment (jglamine) [inline packages/lesswrong/components/common/SearchBar.jsx]: In general some comments on this CSS might be helpful.

File: packages/lesswrong/components/common/SearchBar.jsx [hunk 2]
```diff
@@ -1,23 +1,56 @@
...
+      display: "none"
+    },
+    "& .ais-SearchBox__input": {
+      paddingRight: 0,
+      paddingLeft: 48,
```
Comment (jglamine) [inline packages/lesswrong/components/common/SearchBar.jsx]: Are "px" used as the unit by default?

File: packages/lesswrong/components/search/SearchBarResults.jsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  },
+})
+
+class SearchBarResults extends Component {
+
+  render() {
+    const { classes } = this.props
+
+    return <div className={classes.root}>
+      <Grid container className={classes.searchResults}>
```
Comment (jglamine) [inline packages/lesswrong/components/search/SearchBarResults.jsx]: Do we want to start moving towards using the material-ui grid for layout in general? I noticed that sometimes we use it but sometimes we don't.

File: packages/lesswrong/components/search/SearchBarResults.jsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+              <div className={classes.header}>
+                <Typography variant="display1">Comments</Typography>
+                <Components.SearchPagination />
+              </div>
+              <Configure hitsPerPage={8} />
+              <Hits hitComponent={(props) => <Components.CommentsSearchHit clickAction={this.closeSearch} {...props} />} />
+            </Index>
+          </Grid>
+        </Components.ErrorBoundary>
+        <Components.ErrorBoundary>
```
Comment (jglamine) [inline packages/lesswrong/components/search/SearchBarResults.jsx]: In general I think we kind of go overboard when it comes to error boundaries, but I'm interested to hear your thoughts. How do you go about deciding if there should be an error boundary?

File: packages/lesswrong/components/common/SearchBar.jsx (hunk 2 above)
Comment (Raemon) [reply packages/lesswrong/components/common/SearchBar.jsx]: Yes

File: packages/lesswrong/components/search/SearchBarResults.jsx (hunk 4 above)
Comment (Raemon) [reply packages/lesswrong/components/search/SearchBarResults.jsx]: I asked Habryka and Jim about this at the time, and the idea was "whenever there's a thing that might break independently while leaving other stuff on the page that that the user would derive value from, use an error boundary." In this case, something might break within the posts, users or comments list, and it'd be nice if the other two didn't break also in that case.

File: packages/lesswrong/components/common/SearchBar.jsx (hunk 1 above)
Comment (Raemon) [reply packages/lesswrong/components/common/SearchBar.jsx]: Yeah, I'll add that in. Basically our search components come with weird classes we don't have much direct control over, but which we can change via CSS.

Comment (Raemon) [reply packages/lesswrong/components/common/SearchBar.jsx]: (this isn't meant to be "a way we do things", so much as a particular hack that was worth it in this case)

File: packages/lesswrong/components/common/SearchBar.jsx [hunk 5]
```diff
@@ -69,78 +102,34 @@ class SearchBar extends Component {
...
-    const resultsOpenClass = this.state.searchOpen ? "open" : null;
     const algoliaAppId = getSetting('algolia.appId')
     const algoliaSearchKey = getSetting('algolia.searchKey')
 
+    const { classes } = this.props
     const { searchOpen } = this.state
 
     if(!algoliaAppId) {
-      return <div className="search-bar">Search is disabled (Algolia App ID not configured on server)</div>
+      return <div className={classes.root}>Search is disabled (Algolia App ID not configured on server)</div>
```
Comment (Discordius) [inline packages/lesswrong/components/common/SearchBar.jsx]: This should be `classes.searchBar`, though actually I think the `searchBar` class should be renamed to `root` to comply with the Material UI conventions.

File: packages/lesswrong/components/search/CommentsSearchHit.jsx [hunk 6]
```diff
@@ -2,31 +2,32 @@ import { Components, registerComponent} from 'meteor/vulcan:core';
...
-          <div className="comments-item-date"> {moment(new Date(hit.postedAt)).fromNow()}</div>
-        </div>
-        <div className="recent-comments-item-text comments-item-text">
-          <Snippet attributeName="body" hit={hit} tagName="mark" />
-        </div>
-        </div>
-      </Link>
-  </div>
+const CommentsSearchHit = ({hit, clickAction, classes}) => {
+  return <Link to={"/posts/" + hit.postId + "/" + hit.postSlug + "#" + hit._id} onClick={(event) => isLeftClick(event) && clickAction()} className={classes.root}>
```
Comment (Discordius) [inline packages/lesswrong/components/search/CommentsSearchHit.jsx]: I would split this up into multiple lines.

File: packages/lesswrong/components/common/SearchBar.jsx (hunk 5 above)
Comment (Raemon) [reply packages/lesswrong/components/common/SearchBar.jsx]: Not 100% sure about this (I'm not sure it's technically the root element), but went ahead and changed it to root

**Comment by Raemon** [comment]
have updated the PR

**Pull request merged**